src/Utilities/TrackedDependencies/FlatTrackingData.cs

```diff
@@ -462,7 +484,7 @@ private void ConstructFileTable()
                                 tlogEntry = tlog.ReadLine();
                                 continue;
                             }
-                            else if (tlogEntry[0] == '^' && TreatRootMarkersAsEntries) // This is a rooting record, and we should keep it
+                            else if (tlogEntry[0] == '^' && TreatRootMarkersAsEntries && tlogEntry.IndexOf('|') < 0) // This is a rooting non composite record, and we should keep it
```

Comment:
And we don't need to treat them as entries because for the tools which have them the sources are naturally listed as inputs.